src/Network/Network/FirewallPolicy/PolicySettings/AzureApplicationGatewayFirewallPolicySetting.cs

```diff
@@ -106,8 +128,11 @@ protected PSApplicationGatewayFirewallPolicySettings NewObject()
             {
                 Mode = this.Mode,
                 State = this.State,
+                RequestBodyEnforcement = this.DisableRequestBodyEnforcement != null && (this.DisableRequestBodyEnforcement == true ? false : true),
```

Comment:
Too complicated, lets move this out and set this field below.

Reply:
I think in this PSApplicationGatewayFirewallPolicySettings we cant have fields below since RequestBodyEnforcement is only inside PSApplicationGatewayFirewallPolicySettings

Reply:
You don't need to return right away. You can create a local object, do whatever you want and then return that object.

Reply:
move the condition outside of the return object